sympy/solvers/solvers.py

```diff
@@ -331,6 +331,9 @@ def check_assumptions(expr, **assumptions):
        >>> z = Symbol('z')
        >>> check_assumptions(z, real=True)
     """
+    if e is not None:
```

Comment:
Maybe a better name for this argument would be `against` (i.e., you are checking the assumptions of `expr` *against* the assumptions of `against`).